File: OsmAnd/src/net/osmand/plus/card/color/palette/OnColorsPaletteListener.java
package net.osmand.plus.card.color.palette;

import androidx.annotation.ColorInt;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import net.osmand.plus.card.color.palette.data.PaletteColor;

public interface OnColorsPaletteListener {

	void onColorSelectedFromPalette(@NonNull PaletteColor paletteColor);

	default void onColorAddedToPalette(@Nullable PaletteColor oldColor, @NonNull PaletteColor newColor) {
	}

}

File: OsmAnd/src/net/osmand/plus/card/color/palette/main/v2/OnColorsPaletteListener.kt
package net.osmand.plus.card.color.palette.main.v2

import net.osmand.shared.palette.domain.PaletteItem

interface OnColorsPaletteListener {

	fun onPaletteItemSelected(item: PaletteItem)

	fun onPaletteItemAdded(oldItem: PaletteItem?, newItem: PaletteItem) {
	}
}